pandas/_libs/tslibs/parsing.pyx

```diff
@@ -188,19 +207,27 @@ cdef parse_datetime_string_with_reso(date_string, freq=None, dayfirst=False,
 
 
 cpdef bint _does_string_look_like_datetime(object date_string):
-    if date_string.startswith('0'):
-        # Strings starting with 0 are more consistent with a
-        # date-like string than a number
-        return True
-
-    try:
-        if float(date_string) < 1000:
-            return False
-    except ValueError:
-        pass
+    cdef:
+        char *buf = NULL
+        char *endptr = NULL
+        Py_ssize_t length = -1
```

Comment:
can Py_ssize_t be negative?

Reply:
Yes, first `s` in `ssize` stands for `signed`.